internal/manifest/oci_index.go

```diff
@@ -93,19 +147,51 @@ func (index *OCI1IndexPublic) ChooseInstance(ctx *types.SystemContext) (digest.D
 				Variant:      d.Platform.Variant,
 			}
 			if platform.MatchesPlatform(imagePlatform, wantedPlatform) {
-				return d.Digest, nil
+				if bestMatchedManifest.Digest == "" || instanceBetterMatchesCompression(d, bestMatchedManifest, didPreferGzip) {
```

Comment:
Surprisingly enough, AFAICS nothing here enforces that entries of `manifests` all have `.Digest != ""` (and this function is probably not the place to do that). _Conceptually_, this condition therefore does not quite express the intent. A pedantically more correct alternative would be to have `bestMatch *….Descriptor`, with `nil` as the “no match yet” case.

OTOH, sure, an item with `Digest == ""` would be unusable… and anyway I this is, at best, an item to deal with the last tune-up.

---

(The `bestMatchedManifest.Digest == ""` only needs to be in _one_ of `instanceBetterMatchesCompression` and its caller. I think ideally it should be in the caller, but that makes more sense to make if there is only one call site.)